Sources/ViberUIGridView/UIGrid.swift

```diff
@@ -0,0 +1,241 @@
+        guard !buttons.isEmpty else {
+            throw UIGridError.emptyButtons
+        }
+        
+        return UIGrid(type: .keyboard,
+                      isDefaultHeight: isDefaultHeight,
+                      customDefaultHeight: customDefaultHeight,
+                      buttons: buttons,
+                      buttonsGroupColumns: buttonsGroupColumns,
+                      buttonsGroupRows: buttonsGroupRows,
+                      heightScale: nil,
+                      backgroundColor: backgroundColor,
+                      inputFieldState: inputFieldState,
+                      favoritesMetadata: nil)
+    }
+    
+    public static func rich(with buttons: [UIGridButton],
+                            buttonsGroupColumns: Int? = nil,
+                            buttonsGroupRows: Int? = nil,
+                            backgroundColor: String? = nil) throws -> Self {
+        guard !buttons.isEmpty else {
+            throw UIGridError.emptyButtons
+        }
+        
+        return UIGrid(type: .keyboard,
+                      isDefaultHeight: nil,
+                      customDefaultHeight: nil,
+                      buttons: buttons,
+                      buttonsGroupColumns: buttonsGroupColumns,
+                      buttonsGroupRows: buttonsGroupRows,
+                      heightScale: nil,
+                      backgroundColor: backgroundColor,
+                      inputFieldState: nil,
+                      favoritesMetadata: nil)
+    }
+    
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.type = try container.decodeIfPresent(UIGridType.self, forKey: .type) ?? .unknown
+        self.isDefaultHeight = try container.decodeIfPresent(SomeKindOfBool.self, forKey: .isDefaultHeight)?.wrappedValue
+        self.customDefaultHeight = try container.decodeIfPresent(Int.self, forKey: .customDefaultHeight)
+        guard let buttons = try container.decodeIfPresent([UIGridButton].self, forKey: .buttons),
+              !buttons.isEmpty else {
+                  throw UIGridError.emptyButtons
+              }
+        self.buttons = buttons
+        self.buttonsGroupColumns = try container.decodeIfPresent(Int.self, forKey: .buttonsGroupColumns)
+        self.buttonsGroupRows = try container.decodeIfPresent(Int.self, forKey: .buttonsGroupRows)
+        self.heightScale = try container.decodeIfPresent(Int.self, forKey: .heightScale)
+        self.backgroundColor = try container.decodeIfPresent(String.self, forKey: .backgroundColor)
+        self.inputFieldState = try container.decodeIfPresent(UIGrid.InputFieldState.self, forKey: .inputFieldState)
+        self.favoritesMetadata = try container.decodeIfPresent(UIGridFavorites.self, forKey: .favoritesMetadata)
+    }
+    
+//    public static let keyboard(buttonBuilders: [UIGridButtonBuilder?],
```

Comment:
Either place a TODO what should be done to uncomment this code or simply remove it